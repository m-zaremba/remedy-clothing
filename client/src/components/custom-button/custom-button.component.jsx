import React from 'react';
import PropTypes from 'prop-types';
import { CustomButtonContainer } from './custom-button.styles';

const CustomButton = ({ children, ...props }) => {
  return (
    <CustomButtonContainer type="button" {...props}>
      {children}
    </CustomButtonContainer>
  );
};

CustomButton.defaultProps = {
  isGoogleSignIn: false,
  inverted: false,
};

CustomButton.propTypes = {
  children: PropTypes.node.isRequired,
  isGoogleSignIn: PropTypes.bool,
  inverted: PropTypes.bool,
};

export default CustomButton;
