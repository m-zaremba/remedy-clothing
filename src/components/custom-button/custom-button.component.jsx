/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable react/button-has-type */
import React from 'react';
import './custom-button.styles.scss';
import PropTypes from 'prop-types';

const CustomButton = ({ children, isGoogleSignIn, inverted, ...otherProps }) => {
  return (
    <button
      className={`${inverted ? 'inverted' : ''} ${
        isGoogleSignIn ? 'google-sign-in' : ''
      } custom-button`}
      {...otherProps}
    >
      {children}
    </button>
  );
};

CustomButton.defaultProps = {
  isGoogleSignIn: false,
  inverted: false,
};

CustomButton.propTypes = {
  children: PropTypes.node.isRequired,
  isGoogleSignIn: PropTypes.bool,
  inverted: PropTypes.bool,
};

export default CustomButton;
