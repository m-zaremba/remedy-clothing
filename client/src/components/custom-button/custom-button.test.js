import React from 'react';
import { shallow } from 'enzyme';
import { CustomButton } from './custom-button.component';

describe('testing CustomButton component', () => {
  const mockChildren = <div>test</div>;
  it('should render CustomButton component', () => {
    expect(shallow(<CustomButton>{mockChildren}</CustomButton>)).toMatchSnapshot();
  });
});
