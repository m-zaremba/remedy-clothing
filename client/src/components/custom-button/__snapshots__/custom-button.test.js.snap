// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing CustomButton component should render CustomButton component 1`] = `
<styled.button
  inverted={false}
  isGoogleSignIn={false}
  type="button"
>
  <div>
    test
  </div>
</styled.button>
`;
